lib/python/src/aria/ops/result.py

```diff
@@ -111,6 +115,44 @@ def send_results(self, output_pipe: str = sys.argv[-1]) -> None:
         write_to_pipe(output_pipe, self.get_json())
 
 
+RelationshipUpdateMode = NewType("RelationshipUpdateMode", int)
+
+
+class RelationshipUpdateModes(Enum):  # type: ignore
+    """
+    If 'update_relationships' is 'ALL', all relationships between objects are
+    returned.
+
+    If 'update_relationships' is 'None', no relationships will be returned.
+
+    If 'update_relationships' is 'AUTO' (or not explicitly set), then all
+    relationships are returned _if and only if at least one object has set
+    relationships_.
+
+    If 'update_relationships' is 'PER_OBJECT', then only objects with updated
+    relationships will be returned. Note: Any object that has not been updated
+    will retain existing relationships. This means that to remove all
+    relationships from an object (without setting any new relationships), the
+    adapter must call 'add_children' on the object with an empty collection of
+    children.
+
+    If an object's relationships are returned, but the object has no current
```

Comment:
I'm having trouble parsing this sentence.

Reply:
Hmm, I see - I think we need some better terminology. It turns out that this:

```
relationships: []
```
and this:
```
relationships: [
   {
       parent: { ... }
       children: []
   }
]
```
are not equivalent. The first one returns no relationship objects, and the second returns a relationship object that has no relationships. The first one does nothing, and the second one removes any existing children relationships to the specified parent (if there were any). The sentence is referring to the second case.

Do you have any wording suggestions?

Reply:
much better now!